code/datums/ai/_ai_controller.dm

```diff
@@ -366,6 +366,32 @@ multiple modular subtrees with behaviors
 			ProcessBehavior(action_seconds_per_tick, current_behavior)
 			return
 
+///Can this pawn interact with objects?
+/datum/ai_controller/proc/ai_can_interact()
+	SHOULD_CALL_PARENT(TRUE)
+	return !QDELETED(pawn)
+
+///Interact with objects
+/datum/ai_controller/proc/ai_interact(target, combat_mode, list/modifiers)
+	if(!ai_can_interact())
+		return FALSE
+
+	var/atom/final_target = isdatum(target) ? target : blackboard[target] //incase we got a blackboard key instead
+
+	if(QDELETED(final_target))
+		return FALSE
+
+	var/mob/living/living_pawn = pawn
+	if(isnull(combat_mode))
+		living_pawn.ClickOn(final_target, modifiers)
+		return TRUE
+
+	var/old_combat_mode = living_pawn.combat_mode
+	living_pawn.set_combat_mode(combat_mode)
+	living_pawn.ClickOn(final_target, modifiers)
```

Comment:
You have to refactor click code and replace the `params` argument of `ClickOn` with a list arg. Sure, you could just encapsulate the modifiers list in a `list2params` call instead, but that'd be just bad, because they woud be converted right back into a list right on the sixth line of `ClickOn`...

Reply:
issue is atom/click only takes in params as args and passes them to ClickOn as such. i could params2list them before i pass them on but thatd lead us back to the same bad practice u were describing since theyd need to be converted back into params for many procs that ClickOn calls